src/KnowbaseItem.php

```diff
@@ -888,7 +907,23 @@ public function showFull($options = [])
          */
         global $CFG_GLPI, $DB;
 
-        if (!$this->can($this->fields['id'], READ)) {
+        if (
```

Comment:
Code should probably be after the `default_options` block, as you are adding a new "token" option (which should have its default value set too).

In that case, instead of checking `isset($options['token'])`, we could check `!empty($options['token'])`.

Reply:
If the value is always defined via `default_options`, I don't think the `!empty($options['token'])` check is necessary?